src/test/scala/sam/sceval/XValidatorSpecs.scala

```diff
@@ -16,30 +16,24 @@ class XValidatorSpecs extends Specification with ScalaCheck with IntellijHighlig
   val xvalidator = XValidator(folds = folds, evalBins = Some(4))
   implicit val _: Arbitrary[Int] = Arbitrary(Gen.choose(1, 10))
 
-  // TODO Dry this, write listOfNs method
-  implicit val arbitraryListIntBool: Arbitrary[List[(Int, Boolean)]] = Arbitrary(Gen.frequency(
-    (1, Gen.listOfN(1, arbitrary[(Int, Boolean)])),
-    (1, Gen.listOfN(2, arbitrary[(Int, Boolean)])),
-    (1, Gen.listOfN(4, arbitrary[(Int, Boolean)])),
-    (1, Gen.listOfN(8, arbitrary[(Int, Boolean)])),
-    (1, Gen.listOfN(10, arbitrary[(Int, Boolean)])),
-    (1, Gen.listOfN(15, arbitrary[(Int, Boolean)])),
-    (1, Gen.listOfN(20, arbitrary[(Int, Boolean)])),
-    (1, Gen.listOfN(50, arbitrary[(Int, Boolean)])),
-    (1, Gen.listOfN(100, arbitrary[(Int, Boolean)]))
-  ))
+  // GMARIO: try to add a pattern e.g. 1 to 128 by power of 2 so that we don't have to hardcode the values.
```

Comment:
The lack of power of two is intentional @gm-spacagna  , if we only used powers of two, we could only be sure the code works for powers of two :)